cypress/integration/mhu_kustannuskirjaus.js

```diff
@@ -0,0 +1,129 @@
+import * as ks from '../support/kustannussuunnitelmaFns.js';
+
+let avaaKulujenKohdistus = (urakanNimi) => {
+    cy.visit('/');
+    let hakupalkki = cy.get('.haku-input[placeholder="Hae Harjasta"]')
+    hakupalkki.type(urakanNimi);
+    hakupalkki.siblings().contains(urakanNimi).click();
+    cy.get('[data-cy="tabs-taso1-Kulut"]').click();
+}
+
+let haeUrakanPaatokset = (urakka, hoitokausi) => {
+    return cy.terminaaliKomento().then((tk) => {
+        cy.exec(tk + 'psql -h localhost -U harja harja -c ' +
+            `\"SELECT \\"urakoitsijan-maksu\\" ` + `FROM urakka_paatos ` +
+            `WHERE \\"urakka-id\\" = (SELECT id FROM urakka WHERE nimi = '${urakka}') ` +
+            `AND poistettu = false ` +
+            `AND \\"hoitokauden-alkuvuosi\\" = ${hoitokausi}\"`)
+            .then((dbTulos) => {
+                return Number.parseFloat(dbTulos.stdout.split('\n')[2])
+            })
+    })
+}
+
+let avaaKulunKirjaus = () => {
+    cy.get('button').contains('Uusi kulu').click();
+    cy.contains("Uusi kulu")
+}
+
+let valitseKulunPvm = () => {
+    cy.get('[data-cy="koontilaskun-kk-dropdown"]').within(() => {
+        cy.get('button').click();
+        cy.contains('Syyskuu - 1. hoitovuosi').click();
+    })
+
+    cy.get('.kalenteri-kontti').within(() => {
+        cy.get('input').click();
+        cy.get('td').contains('30').click();
+    })
+}
+
+let tallennaJaTarkistaKulu = (kuluTaiKulut) => {
+    cy.contains('Tallenna').click();
+
+    cy.contains('Kulujen kohdistus')
```

Comment:
Pitäiskö tässä ollu joku tarkistus vielä? Tyyliin .should('exist') tms?

Reply:
contains() kaatuu, mikäli elementtiä ei löydy